### PR TITLE
Patches for 0.59.1

### PR DESCRIPTION
**Pull request by sklam**
- Cherrypick for 0.59.1
- change log
- skipped PPC test failure for `test_issue9490_non_det_ssa_problem`
- 
TODO:
- [ ] cherrypick https://github.com/numba/numba/pull/9480/commits/db03908778713c5ec8c7bc027ae52fa3f409a764  and https://github.com/numba/numba/pull/9480/commits/b39bdf2516174c6f723d6785bd5f70e7a301b4cd for main
- [ ] cherrypick release note update
- [ ] cherrypick towncrier removal

### CODE REVIEW

**Comment by sklam** [comment]
BFID: `numba_yaml_439`

**Review by kc611 — APPROVED** [review]

**Comment by sklam** [comment]
https://github.com/numba/numba/pull/9480#issuecomment-1984329512 

passed. subsequent commits have no code changes.

**Review by kc611 — DISMISSED** [review]

**Comment by sklam** [comment]
Waiting for release blocker fix

**Comment by sklam** [comment]
`numba_yaml_440`

**Comment by sklam** [comment]
BFID: `numba_yaml_441`

**Comment by sklam** [comment]
Note for reviewer:

After cherrypicks and changelog update, the following tests adjustments to `test_issue9490_non_det_ssa_problem` were made:

- skip LLVM codegen failure on PPC for https://github.com/numba/numba/pull/9480/commits/db03908778713c5ec8c7bc027ae52fa3f409a764
```
AssertionError: subprocess failed with output:
Invalid PPC CTR loop!
UNREACHABLE executed at /opt/conda/conda-bld/llvmdev_1684265335202/work/llvm/lib/Target/PowerPC/PPCCTRLoops.cpp:179!
Fatal Python error: Aborted
```
  I believe this is purely a LLVM codegen problem.
- On linux-64 py3.9 npy1.25 , the buildfarm consistently fail the test due to a TBB problem (https://github.com/numba/numba/pull/9480/commits/b39bdf2516174c6f723d6785bd5f70e7a301b4cd). 
  GDB points to:

```
(gdb) info threads
  Id   Target Id                                    Frame
* 1    Thread 0x7ffff7c39280 (LWP 3954038) "python" __pthread_clockjoin_ex (threadid=140736834631424, thread_return=0x0,
    clockid=<optimized out>, abstime=<optimized out>, block=<optimized out>) at pthread_join_common.c:145
  2    Thread 0x7fffdc49c700 (LWP 3954042) "python" syscall () at ../sysdeps/unix/sysv/linux/x86_64/syscall.S:38
  3    Thread 0x7fffdc09b700 (LWP 3954043) "python" 0x00007ffff7d3f89b in sched_yield () at ../sysdeps/unix/syscall-template.S:78
  4    Thread 0x7fffdbc9a700 (LWP 3954044) "python" 0x00007ffff7d3f89b in sched_yield () at ../sysdeps/unix/syscall-template.S:78
  5    Thread 0x7fffdb899700 (LWP 3954045) "python" 0x00007ffff7d3f89b in sched_yield () at ../sysdeps/unix/syscall-template.S:78
  6    Thread 0x7fffdb498700 (LWP 3954047) "python" syscall () at ../sysdeps/unix/sysv/linux/x86_64/syscall.S:38
  7    Thread 0x7fffdac96700 (LWP 3954048) "python" 0x00007ffff7d3f89b in sched_yield () at ../sysdeps/unix/syscall-template.S:78
  8    Thread 0x7fffdb097700 (LWP 3954046) "python" 0x00007ffff7d3f89b in sched_yield () at ../sysdeps/unix/syscall-template.S:78
  12   Thread 0x7fffd9c92700 (LWP 3954054) "python" 0x00007ffff7d3f89b in sched_yield () at ../sysdeps/unix/syscall-template.S:78
  13   Thread 0x7fffd9490700 (LWP 3954053) "python" 0x00007ffff7d3f89b in sched_yield () at ../sysdeps/unix/syscall-template.S:78
  14   Thread 0x7fffda895700 (LWP 3954050) "python" 0x00007ffff7d3f89b in sched_yield () at ../sysdeps/unix/syscall-template.S:78
  15   Thread 0x7fffd908f700 (LWP 3954055) "python" 0x00007ffff5b4349a in __kmp_acquire_ticket_lock ()
   from /home/siu/miniconda3/envs/testenv_bad/lib/python3.9/site-packages/mkl/../../../libiomp5.so
```
  I believe this is bad interaction between TBB and MKL OMP on a machine that has 72 cpu cores. The resolution is to force the numba code to use only 1-thread.

File: docs/source/user/installing.rst [hunk 1]
```diff
@@ -262,6 +262,8 @@ information.
 +----------++--------------+---------------------------+----------------------------+------------------------------+-------------------+-----------------------------+
 | Numba     | Release date | Python                    | NumPy                      | llvmlite                     | LLVM              | TBB                         |
 +===========+==============+===========================+============================+==============================+===================+=============================+
+| 0.59.1    | 2024-03-11   | 3.9.x <= version < 3.13   | 1.22 <= version < 1.27     | 0.42.x                       | 14.x              | 2021.6 <= version           |
```
Comment (kc611) [inline docs/source/user/installing.rst:265]:
```suggestion
| 0.59.1    | 2024-03-18   | 3.9.x <= version < 3.13   | 1.22 <= version < 1.27     | 0.42.x                       | 14.x              | 2021.6 <= version           |
```

**Review by kc611 — APPROVED** [review]

**Pull request merged**